app/assets/javascripts/discourse/app/lib/dag.js

```diff
@@ -25,13 +21,27 @@ export default class DAG {
    * @param {string | string[]} position.before A key or array of keys of items which should appear before this one.
    * @param {string | string[]} position.after A key or array of keys of items which should appear after this one.
    */
-  add(key, value, { before, after } = {}) {
-    // if defaultFirstPosition is set, and no before/after is specified, use the first item as before
-    if (!before && !after && this.defaultFirstPosition) {
-      before = this.firstItemKey;
-    }
-    this.#rawData.set(key, { value, before, after });
-    this.#dag.add(key, value, before, after);
+  add(key, value, position = this.#defaultPosition) {
```

Comment:
How about we split out the default resolution logic into a separate private function:
```
#defaultPositionForKey(key){
  const pos = {...this.#defaultPosition};
  if(Array(pos.before).includes(key)){
    delete pos.before;
  }
  if(Array(pos.after).includes(key){
    delete pos.after;
  }
}
```

```js
add(key, value, position){
  position ||= this.#defaultPositionForKey(key);
```